### PR TITLE
Add sitemap generator

### PR DESCRIPTION
**Pull request by hungmi**
This PR adds sitemap_generator and add some links mentioned in https://github.com/joemasilotti/railsdevs.com/issues/40.

I also add `/developers/new` & `/businesses/new` since they are part of sign up process. I can remove them If that is not needed.

I have tested it locally and the sitemap.xml seems to be ok.
But I don't know how to write test for it 😅. Does anyone have any suggestion?

And the next step is set `rake "-s sitemap:refresh"`(according to the [doc](https://github.com/kjvarga/sitemap_generator#crontab)) on Heroku scheduler.

### Pull request checklist

- [x] Your code contains tests relevant for code you modified
- [x] You have linted and tested the project with `bin/check`

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -47,5 +47,5 @@ gem "pundit", "~> 2.1"
 gem "redcarpet", "~> 3.5"
 gem "sidekiq", "~> 6.3"
 gem "stripe", ">= 2.8", "< 6.0"
-
+gem "sitemap_generator"
```
Comment (joemasilotti) [inline Gemfile:50]: Please move this one line above and sort the chunk of gems.

File: config/sitemap.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Set the host name for URL creation
```
Comment (joemasilotti) [inline config/sitemap.rb:1]: Please remove all comments from this file.

File: config/sitemap.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  #
+  #   add articles_path, :priority => 0.7, :changefreq => 'daily'
+  #
+  # Add all articles:
+  #
+  #   Article.find_each do |article|
+  #     add article_path(article), :lastmod => article.updated_at
+  #   end
+
+  add developers_path, changefreq: "always", priority: 1, lastmod: Developer.order(id: :desc).first.try(:updated_at)
```
Comment (joemasilotti) [inline config/sitemap.rb:28]: `Developer.maximum(:updated_at)` will return the most recently updated timestamp and can be used for the `lastmod:` option.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Thanks for kicking this off! I made a few small change requests, other than that it looks good.

We will have to host the sitemap somewhere else, like S3, as Heroku doesn't have persistent storage. But I can take care of that.

**Comment by joemasilotti** [comment]
I'm fine with a single smoke test if it doesn't take too long. Running the command and verifying that the XML file is created without any raised errors should be good enough!

File: Gemfile (hunk 1 above)
Comment (hungmi) [reply Gemfile:49]: Updated.
 I thought these names are started with **s**. 🤣

File: config/sitemap.rb (hunk 2 above)
Comment (hungmi) [reply config/sitemap.rb]: Done!

File: config/sitemap.rb (hunk 3 above)
Comment (hungmi) [reply config/sitemap.rb]: Updated!

**Comment by hungmi** [comment]
I have added a test to run `rails sitemap:refresh:no_ping`.
But I can't find a way to add `-s` when I invoke it, so I followed [this open issue](https://github.com/kjvarga/sitemap_generator/issues/332) to prevent unneccessarily outputs.
Thanks for your review!

**Comment by joemasilotti** [comment]
This is looking great. I'm just figuring out the S3 hosting and will merge in.

**Review by joemasilotti — APPROVED** [review]

**Comment by joemasilotti** [comment]
OK, I think we're good! Turned out I need to add a bunch of stuff to get the S3 upload working and Google Search Console pointing to the sitemap correctly. I'm going to document all of the non-code stuff on the Wiki soon.

Thanks again for the PR!

**Comment by joemasilotti** [comment]
And we're good! Thanks again 🎉

image

**Pull request merged**